app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -11,7 +11,7 @@ class DrawingViewModel : ViewModel() {
     @ColorInt
     var primaryColor = Color.BLACK
 
-    val layerNames: MutableList<String> = mutableListOf()
+    val layerNames: MutableList<LayerViewModel> = mutableListOf()
```

Comment:
You're bleeding `ViewModel` outside the recyclerview use case. Just name this `Layer` if that's the case, and update the name to be `layers`. And letting this be public lets anyone you are passing it to have edit access to it, which seems dangerous

Reply:
changed back to string, your right it is out of scope of ticket. handled everything with this as an string list. In the future it would probably need to be a layer list of Layer objects.